### PR TITLE
LSNBLDR-633: Restrict editing Lessons pages/subpages to one or more people

### PR DESCRIPTION
**Pull request by nicholaswilson100**
This PR is a copy of https://github.com/sakaiproject/sakai/pull/3384 which is not mine but an ex-colleague's that I've rebased as it had merge conflicts

### CODE REVIEW

**Comment by buckett** [comment]
@clhedrick Are you able to look over this?

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java [hunk 1]
```diff
@@ -4484,6 +4487,43 @@ private void createEditTitleDialog(UIContainer tofill, SimplePage page, SimplePa
...
+						String displayName = user.getDisplayName();
+						if(StringUtils.isBlank(displayName)){
+							displayName = user.getDisplayId();
+						}
+						roleLabels.add(displayName);
+
+					}
+				}
+			} catch (GroupNotDefinedException e) {
+				//TODO add logger warning
```
Comment (buckett) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java]: Is this an actual TODO?

**Comment by clhedrick** [comment]
Yes. I’m about to put it into my copy.

On Nov 15, 2016, at 7:04 AM, Matthew Buckett <notifications@github.com<mailto:notifications@github.com>> wrote:


@clhedrick<https://github.com/clhedrick> Are you able to look over this?

**Comment by clhedrick** [comment]
I tried it. I can't set an owner because the pulldown for possible owners doesn't have anyone.
I can't figure out what's supposed to be there. the template is 

<option rsf:id="possOwner:"></option>

But no java file uses possOwner. It looks like something is missing. Is this patch intended to go with another patch?

**Comment by clhedrick** [comment]
OK, so I see you have to give them a role of "contribute." This seems non-standard. I'd expect you to add a new permission. Your site might well choose to create a standard role of contribute with that permission, but I don't think we want to hardcode an English role name and attach permission to it.

**Comment by clhedrick** [comment]
OK, aside from the hardcoded role, the functionality looks fine. Now I'm going to review the code to see if there are any issues with implementation.

**Comment by clhedrick** [comment]
I the page properties dialog, the choose page owner should be initialized with the current owner.
Also, I'd like the owner name shown both on the super-page after the title of the subpage and on the page itself. Probably not for students, but at least for the site owner and probably for the page owner.

**Comment by clhedrick** [comment]
I set a page owner for the top-level page. There doesn't seem to be a way to unset it.

**Comment by clhedrick** [comment]
Consider whether you want to allow changing owner of a top-level page. If you do, even the site owner no longer gets the "more tools" menu. I think they should always get it.

**Comment by clhedrick** [comment]
In OrphanPageFinder.java, there's a call to getowner that probably needs an additional check whether it's a student page. We don't show student pages as orphans, but you probably do want to show owned subpages.

**Comment by clhedrick** [comment]
I've finished my pass over the code, so if you fix the issues above, I'll accept the patch.

**Comment by clhedrick** [comment]
Ugh. You're not going to like this one.
On a student page, if someone uploads content it's uploaded into the student's worksite. They theory is that we want to charge it against their quota, and students won't generally have permission to load stuff into the site's content. This code is in  getCollectionId in SimplePageBean.

Generally other users can't see files in someone's worksite. So LessonBuilderAccessService.java ignores normal access controls, and (in theory -- I'm checking the code over the next couple of days) allows those who can see a page to see all the files included on it. Because student pages don't have required items, it doesn't bother to invoke all the availabiity stuff. If the person trying to view the site is in the site (or in the group if there's group access involved) we let them see any file on a student page.

I'm not sure what you want to happen on the user-owned pages. Do you assume that the contribute role can create files in the site content? By default Lessons creates a folder for uploaded content. Should it create a folder per contribute-user? Or should it put stuff in the contribute user's worksite the way it does for student pages? Either way, I believe pages contributed by a contribute user can use accessibility restrictions, so those restrictions will need to be implemented in LessonBuilderAccessService.java. All the right stuff is there in LessonBuilderAccessService.java. It's just a matter of the right if tests, depending upon what you think should happen. If you're willing to give the contribute role write access to the site content, you could just tweak a couple of tests in getCollectionId and LessonBuilderAccessService to check for only actual student pages rather than owned pages. But if you want something more complex, more code may be needed.

**Comment by nicholaswilson100** [comment]
Hey @clhedrick,

After some discussion here at Oxford with Adam, Matthew and others, I've updated this Jira so it's a bit different and has some new requirements (which will change Neelam's code substantively).  When you have a spare second, could you let me know what you think? https://jira.sakaiproject.org/browse/LSNBLDR-633

**Pull request closed without merging**